src/test/java/org/elasticsearch/indices/exists/indices/IndicesExistsTests.java

```diff
@@ -0,0 +1,85 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.indices.exists.indices;
+
+import org.elasticsearch.action.admin.indices.settings.put.UpdateSettingsResponse;
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.cluster.block.ClusterBlockException;
+import org.elasticsearch.cluster.metadata.IndexMetaData;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.test.ElasticsearchIntegrationTest;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.elasticsearch.common.settings.ImmutableSettings.settingsBuilder;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.notNullValue;
+
+public class IndicesExistsTests extends ElasticsearchIntegrationTest {
+
+    @Test
+    // Indices exists never throws IndexMissingException, the indices options control its behaviour (return true or false)
+    public void testIndicesExists() throws Exception {
+        assertThat(client().admin().indices().prepareExists("foo").get().isExists(), equalTo(false));
+        assertThat(client().admin().indices().prepareExists("foo").setIndicesOptions(IndicesOptions.lenientExpandOpen()).get().isExists(), equalTo(true));
+        assertThat(client().admin().indices().prepareExists("foo*").get().isExists(), equalTo(false));
+        assertThat(client().admin().indices().prepareExists("foo*").setIndicesOptions(IndicesOptions.fromOptions(false, true, true, false)).get().isExists(), equalTo(true));
+        assertThat(client().admin().indices().prepareExists("_all").get().isExists(), equalTo(false));
+
+        createIndex("foo", "foobar", "bar", "barbaz");
+        ensureYellow();
+
+        assertThat(client().admin().indices().prepareExists("foo*").get().isExists(), equalTo(true));
+        assertThat(client().admin().indices().prepareExists("foobar").get().isExists(), equalTo(true));
+        assertThat(client().admin().indices().prepareExists("bar*").get().isExists(), equalTo(true));
+        assertThat(client().admin().indices().prepareExists("bar").get().isExists(), equalTo(true));
+        assertThat(client().admin().indices().prepareExists("_all").get().isExists(), equalTo(true));
+    }
+
+    @Test
+    public void testIndicesExistsWithBlocks() {
+        List<String> blocks = Arrays.asList(IndexMetaData.SETTING_BLOCKS_READ,
+                                            IndexMetaData.SETTING_BLOCKS_WRITE,
+                                            IndexMetaData.SETTING_BLOCKS_METADATA);
+        createIndex("ro");
+        ensureYellow();
+
+        for (String blockSetting : blocks) {
+            try {
+                setIndexBlocks("ro", blockSetting, true);
+                assertThat(client().admin().indices().prepareExists("ro").execute().actionGet().isExists(), equalTo(true));
+            } catch (ClusterBlockException e) {
```

Comment:
I think we can omit catching this and failing when caught, that's default behaviour, what matters if the `finally` I guess